src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/parser/FinanceTrackerParserTest.java

```diff
@@ -33,102 +33,501 @@
+                instanceof ListExpenseCommand);
+        assertTrue(parser.parseCommand(ListExpenseCommand.COMMAND_WORD + " 3",
+                incomeUiStateStub) instanceof ListExpenseCommand);
     }
 
     @Test
-    public void parseCommand_listIncome() throws Exception {
-        assertTrue(parser.parseCommand(ListIncomeCommand.COMMAND_WORD) instanceof ListIncomeCommand);
-        assertTrue(parser.parseCommand(ListIncomeCommand.COMMAND_WORD + " 3") instanceof ListIncomeCommand);
+    public void parseCommand_listExpenseWhenExpensesTab() throws Exception {
+        assertTrue(parser.parseCommand(ListExpenseCommand.COMMAND_WORD, overviewUiStateStub)
+                instanceof ListExpenseCommand);
+        assertTrue(parser.parseCommand(ListExpenseCommand.COMMAND_WORD + " 3",
+                expensesUiStateStub) instanceof ListExpenseCommand);
     }
 
     @Test
-    public void parseCommand_unrecognisedInput_throwsParseException() {
+    public void parseCommand_listExpenseWhenAnalyticsTab() throws Exception {
+        assertTrue(parser.parseCommand(ListExpenseCommand.COMMAND_WORD, overviewUiStateStub)
+                instanceof ListExpenseCommand);
+        assertTrue(parser.parseCommand(ListExpenseCommand.COMMAND_WORD + " 3",
+                analyticsUiStateStub) instanceof ListExpenseCommand);
+    }
+
+    @Test
+    public void parseCommand_listIncomeWhenOverviewTab() throws Exception {
+        assertTrue(parser.parseCommand(ListIncomeCommand.COMMAND_WORD, overviewUiStateStub)
+                instanceof ListIncomeCommand);
+        assertTrue(parser.parseCommand(ListIncomeCommand.COMMAND_WORD + " 3",
+                overviewUiStateStub) instanceof ListIncomeCommand);
+    }
+
+    @Test
+    public void parseCommand_listIncomeWhenIncomeTab() throws Exception {
+        assertTrue(parser.parseCommand(ListIncomeCommand.COMMAND_WORD, overviewUiStateStub)
+                instanceof ListIncomeCommand);
+        assertTrue(parser.parseCommand(ListIncomeCommand.COMMAND_WORD + " 3",
+                incomeUiStateStub) instanceof ListIncomeCommand);
+    }
+
+    @Test
+    public void parseCommand_listIncomeWhenExpensesTab() throws Exception {
+        assertTrue(parser.parseCommand(ListIncomeCommand.COMMAND_WORD, overviewUiStateStub)
+                instanceof ListIncomeCommand);
+        assertTrue(parser.parseCommand(ListIncomeCommand.COMMAND_WORD + " 3",
+                expensesUiStateStub) instanceof ListIncomeCommand);
+    }
+
+    @Test
+    public void parseCommand_listIncomeWhenAnalyticsTab() throws Exception {
+        assertTrue(parser.parseCommand(ListIncomeCommand.COMMAND_WORD, overviewUiStateStub)
+                instanceof ListIncomeCommand);
+        assertTrue(parser.parseCommand(ListIncomeCommand.COMMAND_WORD + " 3",
+                analyticsUiStateStub) instanceof ListIncomeCommand);
+    }
+
+    @Test
+    public void parseCommand_unrecognisedInputWhenOverviewTab_throwsParseException() {
```

Comment:
```suggestion
    public void parseCommand_unrecognizedInputWhenOverviewTab_throwsParseException() {
```
Also in other similar places.